openmetadata-ui/src/main/resources/ui/src/pages/RolesPage/AddRuleModal.tsx

```diff
@@ -0,0 +1,175 @@
+  );
+  const [isEnabled, setIsEnabled] = useState<boolean>(
+    Boolean(initialData.enabled)
+  );
+  const onChangeHadler = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>
+  ) => {
+    e.persist();
+    setData((prevState) => {
+      return {
+        ...prevState,
+        [e.target.name]: e.target.value,
+      };
+    });
+  };
+
+  const onSubmitHandler = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    onSave({
+      ...data,
+      allow: access === RuleAccess.DENY,
+      enabled: isEnabled,
+    });
+  };
+
+  return (
+    <dialog className="tw-modal" data-testid="modal-container">
+      <div className="tw-modal-backdrop" onClick={() => onCancel()} />
+      <div className="tw-modal-container tw-overflow-y-auto tw-max-h-screen">
+        <form onSubmit={onSubmitHandler}>
+          <div className="tw-modal-header">
+            <p
+              className="tw-modal-title tw-text-grey-body"
+              data-testid="header">
+              {header}
+            </p>
+          </div>
+          <div className="tw-modal-body">
+            <div className="tw-mb-4">
+              <label className="tw-form-label required-field">Name</label>
+              <input
+                autoComplete="off"
+                className="tw-text-sm tw-appearance-none tw-border tw-border-main
+                tw-rounded tw-w-full tw-py-2 tw-px-3 tw-text-grey-body  tw-leading-tight
+                focus:tw-outline-none focus:tw-border-focus hover:tw-border-hover tw-h-10"
+                name="name"
+                placeholder="Name"
+                type="text"
+                value={data.name}
+                onChange={onChangeHadler}
+              />
+              {errorData?.name && errorMsg(errorData.name)}
+            </div>
+            <div className="tw-grid tw-grid-cols-2 tw-gap-2">
+              {!isUndefined(initialData.operation) && (
+                <div className="tw-mb-4">
+                  <label className="tw-form-label required-field">
+                    Select Operation
+                  </label>
+                  <select
+                    required
+                    className="tw-text-sm tw-appearance-none tw-border tw-border-main
+                tw-rounded tw-w-full tw-py-2 tw-px-3 tw-text-grey-body  tw-leading-tight
+                focus:tw-outline-none focus:tw-border-focus hover:tw-border-hover tw-h-10 tw-bg-white"
+                    name="operation"
+                    value={data.operation}
+                    onChange={onChangeHadler}>
+                    <option value={Operation.UpdateDescription}>
+                      Update Description
+                    </option>
+                    <option value={Operation.UpdateOwner}>Update Owner</option>
+                    <option value={Operation.UpdateTags}>Update Tags</option>
+                    <option value={Operation.SuggestDescription}>
```

Comment:
For operations, can we expose only the following 3 for now:

```
UpdateOwner
UpdateDescription
UpdateTags
```

The others with `Suggest*` are not hooked up in the backend in any form, since we do not have the feature for users to make suggestions in OpenMetadata frontend/backend yet. 


cc: @harshach something that we need to for upcoming release(s)